modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -537,6 +537,60 @@ void ONNXImporter::populateNet(Net dstNet)
         {
             replaceLayerParam(layerParams, "size", "local_size");
         }
+        else if (layer_type == "InstanceNormalization")
+        {
+            if (node_proto.input_size() != 3)
+                CV_Error(Error::StsNotImplemented,
+                         "Expected input, scale, bias");
+
+            layerParams.blobs.resize(2);
+
+            if (!node_proto.input(1).empty()) {
+                layerParams.blobs.resize(layerParams.blobs.size() + 1);
+                layerParams.set("has_weight", true);
+                layerParams.blobs.back() = getBlob(node_proto, constBlobs, 1);  // weightData
+            } else {
+                layerParams.set("has_weight", false);
+            }
+
+            if (!node_proto.input(2).empty()) {
+                layerParams.blobs.resize(layerParams.blobs.size() + 1);
+                layerParams.set("has_bias", true);
+                layerParams.blobs.back() = getBlob(node_proto, constBlobs, 2);  // biasData
+            } else
+                layerParams.set("has_bias", false);
+
+            Mat mean, std;
+            mean = Mat::zeros(layerParams.blobs[2].total(), 1, CV_32F);
+            std = Mat::ones(layerParams.blobs[2].total(), 1, CV_32F);
```

Comment:
`layerParams.blobs[2]` might be out of range.

Reply:
Hey @dkurt it should be fixed now. Thanks for all yout help!! :smiley: